app/services/badge_service.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+class BadgeService
+  cattr_reader :rules
+
+  @@rules = {
+    pass_all_categories: BaseRule.init(AllCategoriesRule),
```

Comment:
```suggestion
    pass_all_categories: AllCategoriesRule.new,
```
зачем этот хитрый заход через суперкласс?

Reply:
Я не мог определиться как именно реализовывать правила, думал сделать фабрику, но тогда правила хранились бы в BaseRule, что вроде как правильно...  Или нет? Но тк ты мне указал на то что правила должны быть в модели сервиса - и  я сделал по другому, в итоге класс BaseRule оказался не нужен.